cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -207,25 +244,22 @@ private AuthenticationClient getAuthenticationClient(ConnectionConfig connection
     return authenticationClient;
   }
 
-  private AccessToken getSavedAccessToken(String hostname) {
+  private UserAccessToken getSavedAccessToken(String hostname) {
     File file = getAccessTokenFile(hostname);
-    if (file.exists() && file.canRead()) {
-      try {
-        String tokenString = Joiner.on("").join(Files.readLines(file, Charsets.UTF_8));
-        return new AccessToken(tokenString, -1L, null);
-      } catch (IOException ignored) {
-        // Fall through
-      }
+    try (BufferedReader reader = Files.newReader(file, Charsets.UTF_8)) {
+      return GSON.fromJson(reader, UserAccessToken.class);
+    } catch (IOException ignored) {
```

Comment:
Add `JsonSyntaxException` to this block. Otherwise we'll run into issues if we try to read an old version of the access token.